### PR TITLE
Change search background

### PR DESCRIPTION
**Pull request by CsatariGergely**
This pr adds the possibility to change the background
of the search area.

Signed-off-by: Gergely Csatari <gergely.csatari@nokia.com>

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ module.exports = function(grunt) {
     usebanner: {
       dist: {
         options: {
-          position: 'top',
+         position: 'top',
```
Comment (jessetan) [inline Gruntfile.js]: Missing space, please revert this change

File: Gruntfile.js [hunk 2]
```diff
@@ -139,6 +139,9 @@ module.exports = function(grunt) {
       bower_update: {
         cmd: 'bower update'
       },
+      pip_install: {
```
Comment (jessetan) [inline Gruntfile.js]: This is not related to the functional changes in the PR, please revert (also the changes to grunt.registerTask below)

In fact, doing a pip install should not be necessary for using the dev version of the theme with the docs.

File: sphinx_rtd_theme/theme.conf [hunk 3]
```diff
@@ -10,8 +10,9 @@ collapse_navigation = True
 sticky_navigation = True
 navigation_depth = 4
 includehidden = True
+logo_background_color =
```
Comment (jessetan) [inline sphinx_rtd_theme/theme.conf]: I think nav_header_background_color, sidebar_header_background_color or something like it would be better. This is more than just the background for the logo (also contains version and search) and there might not even be a logo

File: Gruntfile.js (hunk 1 above)
Comment (CsatariGergely) [reply Gruntfile.js]: okay

File: Gruntfile.js (hunk 2 above)
Comment (CsatariGergely) [reply Gruntfile.js]: I agree, that it does not related to the functionality of the PR, so I revert it. 

For me my changes were not effective until a pip install --upgrade . , but maybe it is something with my environment.

File: sphinx_rtd_theme/theme.conf (hunk 3 above)
Comment (CsatariGergely) [reply sphinx_rtd_theme/theme.conf]: Okay, I've changed it to nav_header_background_color

**Comment by davidfischer** [comment]
Is this the same as #404?

File: sphinx_rtd_theme/theme.conf [hunk 4]
```diff
@@ -10,8 +10,9 @@ collapse_navigation = True
 sticky_navigation = True
 navigation_depth = 4
 includehidden = True
+nav_header_background_color = 
 titles_only = 
-logo_only =
+logo_only =
```
Comment (jessetan) [inline sphinx_rtd_theme/theme.conf]: Please remove trailing space

**Comment by jessetan** [comment]
Assuming there won't be many of these style overrides, I like this approach more than the one in #404.

Two things I'm not certain on:
- Do we need to prefix this option with something like `style_` so that we can group future style overrides, or do we think we will not have any more?
- This allows changing of the CSS `background-color` property. Would CSS `background` be a better choice so one can do background images or gradients as well?

**Comment by Blendify** [comment]
> Assuming there won't be many of these style overrides, I like this approach more than the one in #404.
> 
> Two things I'm not certain on:
> 
> * Do we need to prefix this option with something like `style_` so that we can group future style overrides, or do we think we will not have any more?
> * This allows changing of the CSS `background-color` property. Would CSS `background` be a better choice so one can do background images or gradients as well?

1. Yes I think we should prefix this.
2. I think so, it gives more freedom.

**Comment by CsatariGergely** [comment]
Thanks for the suggestions @jessetan and @Blendify I've applied both of them.

**Comment by jessetan** [comment]
Code looks OK, just needs some documentation here: https://github.com/rtfd/sphinx_rtd_theme/blob/master/docs/configuring.rst#html-theme-options

**Comment by CsatariGergely** [comment]
@jessetan documentation is added.

File: docs/configuring.rst [hunk 5]
```diff
@@ -25,6 +25,7 @@ file of this repository, and can be defined in your project's ``conf.py`` via
         'prev_next_buttons_location': 'bottom',
         'style_external_links': False,
         'vcs_pageview_mode': '',
+        'style_nav_header_background': '',
```
Comment (jessetan) [inline docs/configuring.rst]: If you are putting this in an example, better to include an actual value.

**Comment by jessetan** [comment]
LGTM

**Comment by CsatariGergely** [comment]
What else is needed to merge this?

**Comment by CsatariGergely** [comment]
@ericholscher , @agjohnson can you please take a look please?

**Comment by ericholscher** [comment]
This feels oddly specific. Is there a reason we'd only support this one setting for a background color? It feels like having the theme be more overridable in general is a good goal, but this is a half measure. The comments on #404 touch on this, but having a full set of colors that can be overridden feels like the proper solution.

**Comment by ericholscher** [comment]
On second thought, this is a good approach for now, but we should do a larger rethink in #404.

**Comment by CsatariGergely** [comment]
Thanks @ericholscher . We can remove this approach once #404 is merged and use a more generic one.
